### PR TITLE
fix(ts): do not clobber spelloptions

### PR DESCRIPTION
**Pull request by lewis6991**

### CODE REVIEW

File: runtime/lua/vim/treesitter/highlighter.lua [hunk 1]
```diff
@@ -97,7 +97,13 @@ function TSHighlighter.new(tree, opts)
   if vim.g.syntax_on ~= 1 then
     vim.api.nvim_command('runtime! syntax/synload.vim')
   end
-  vim.bo[self.bufnr].spelloptions = 'noplainbuffer'
+
+  -- vim.cmd'set spelloptions+=noplainbuffer'
```
Comment (clason) [inline runtime/lua/vim/treesitter/highlighter.lua:101]: `vim.opt.spelloptions:append('noplainbuffer')`

Comment (lewis6991) [reply runtime/lua/vim/treesitter/highlighter.lua:101]: The only reason I didn't go for that is that you can't pass a buffer handle to `vim.opt_local`.

Comment (clason) [reply runtime/lua/vim/treesitter/highlighter.lua:101]: Let's try it out -- if it's needed, it should be added to `vim.opt_local`.

File: runtime/lua/vim/treesitter/highlighter.lua [hunk 2]
```diff
@@ -97,7 +97,10 @@ function TSHighlighter.new(tree, opts)
   if vim.g.syntax_on ~= 1 then
     vim.api.nvim_command('runtime! syntax/synload.vim')
   end
-  vim.bo[self.bufnr].spelloptions = 'noplainbuffer'
+
+  a.nvim_buf_call(self.bunr, function()
```
Comment (clason) [inline runtime/lua/vim/treesitter/highlighter.lua:101]:
```suggestion
  a.nvim_buf_call(self.bufnr, function()
```

Comment (lewis6991) [reply runtime/lua/vim/treesitter/highlighter.lua:101]: ffs

File: runtime/lua/vim/treesitter/highlighter.lua [hunk 3]
```diff
@@ -97,7 +97,10 @@ function TSHighlighter.new(tree, opts)
   if vim.g.syntax_on ~= 1 then
     vim.api.nvim_command('runtime! syntax/synload.vim')
```
Comment (smjonas) [inline runtime/lua/vim/treesitter/highlighter.lua:98]: Might want to change that to `a.nvim_command` here as well while you're at it :)

Comment (vigoux) [reply runtime/lua/vim/treesitter/highlighter.lua:98]: That could even be `vim.cmd`

File: runtime/lua/vim/treesitter/highlighter.lua [hunk 4]
```diff
@@ -95,9 +95,12 @@ function TSHighlighter.new(tree, opts)
   -- syntax FileType autocmds. Later on we should integrate with the
   -- `:syntax` and `set syntax=...` machinery properly.
   if vim.g.syntax_on ~= 1 then
-    vim.api.nvim_command('runtime! syntax/synload.vim')
+    vim.cmd.runtime({ 'syntax/synload.vim', bang = true })
```
Comment (vigoux) [inline runtime/lua/vim/treesitter/highlighter.lua:98]:
```suggestion
    vim.cmd.runtime { 'syntax/synload.vim', bang = true }
```

Comment (lewis6991) [reply runtime/lua/vim/treesitter/highlighter.lua:98]: I would prefer that too, but it goes against our linter rules 😢

**Review by clason — APPROVED** [review]

**Pull request merged**